### PR TITLE
flexynesis GTN - unsupervised task 

### PR DESCRIPTION
**Pull request by nilchia**
This PR adds the training material for unsupervised analysis with flexynesis

### CODE REVIEW

File: topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+> In this tutorial, we will cover:
+>
+> 1. TOC
+> {:toc}
+>
+{: .agenda}
+
+# Data upload
+
+In the first part of this tutorial we will upload processed CITE-seq data from bone marrow tissue.
```
Comment (pavanvidem) [inline topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md:60]: Please elaborate on how many, what information they contain and what formats they are. It will be useful because you will some info from clin files ( such as celltype) in later steps..

File: topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+>    ```
+>
+>    {% snippet faqs/galaxy/datasets_import_via_link.md %}
+>
+> 3. Rename the datasets
+> 4. Check that the datatype is `tabular`
+>
+>    {% snippet faqs/galaxy/datasets_change_datatype.md datatype="tabular" %}
+>
+> 5. Add to each database a tag corresponding to modality (RNA, ADT)
```
Comment (pavanvidem) [inline topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md]: what about the files with "clin"? should we also add tags? btw, is it "dataset" instead of database?

File: topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+>        - *"Number of nearest neighbors to connect for each node"*: `15`
+>
+{: .hands_on}
+> <question-title></question-title>
+>
+> 1. What is the output of this tool?
+>
+> > <solution-title></solution-title>
+> >
+> > 1. The output is the `test-clin_BMscRNAseq.tabular` file with a column added containing Louvain clustering values,
```
Comment (pavanvidem) [inline topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md]:
```suggestion
> > 1. The output is the `test-clin_BMscRNAseq.tabular` file with a column added containing Louvain clustering values.
```

File: topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+> >
+> {: .solution}
+>
+{: .question}
+
+## Get optimal clusters
+We can also use k-means clustering with a varying number of expected clusters and pick the best one based on silhouette scores.
```
Comment (pavanvidem) [inline topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md]:
```suggestion
Now we will use k-means clustering with a varying number of expected clusters and pick the best one based on silhouette scores.
```

File: topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+- Variational autoencoders can effectively capture cellular heterogeneity in single-cell data without requiring labeled training data
+- Flexynesis provides a structured framework for multi-modal data integration with rigorous evaluation procedures
+- Unsupervised feature learning can reveal biologically meaningful cellular populations and relationships
+- Low-dimensional embeddings from VAEs can be used for clustering, visualization, and biological interpretation
+contributors:
+- Nilchia
+- bgruening
+answer_histories:
+    - label: "usegalaxy.eu"
+      history: https://usegalaxy.eu/u/nilchia/h/final-unsupervised-analysis-of-bone-marrow-cells-test-wf
```
Comment (pavanvidem) [inline topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md:26]: Please archive the history so that you can't make any accidental changes to the history in the future.

File: topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+
+## Get optimal clusters
+We can also use k-means clustering with a varying number of expected clusters and pick the best one based on silhouette scores.
+
+> <hands-on-title> Get optimal clusters </hands-on-title>
+>
+> 1. {% tool [Flexynesis utils](toolshed.g2.bx.psu.edu/repos/bgruening/flexynesis_utils/flexynesis_utils/0.2.20+galaxy3) %} with the following parameters:
+>    - *"I certify that I am not using this tool for commercial purposes."*: `Yes`
+>    - *"Flexynesis utils"*: `Get Optimal Clusters`
+>        - {% icon param-file %} *"Predicted labels"*: `louvain_clustering` (output of **Flexynesis utils** {% icon tool %})
```
Comment (pavanvidem) [inline topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md:195]: what about the "Matrix" param?

File: topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md [hunk 7]
```diff
@@ -0,0 +1,303 @@
...
+>        - *"Minimum number of clusters to try"*: `10`
+>        - *"Maximum number of clusters to try"*: `20`
```
Comment (pavanvidem) [inline topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md]: how to choose these values? There were only 9 clusters found in the previous step. is there any relationship between the number of louvain clusters and these values?

File: topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md [hunk 8]
```diff
@@ -0,0 +1,303 @@
...
+{: .question}
+
+## Get optimal clusters
+We can also use k-means clustering with a varying number of expected clusters and pick the best one based on silhouette scores.
+
+> <hands-on-title> Get optimal clusters </hands-on-title>
+>
+> 1. {% tool [Flexynesis utils](toolshed.g2.bx.psu.edu/repos/bgruening/flexynesis_utils/flexynesis_utils/0.2.20+galaxy3) %} with the following parameters:
+>    - *"I certify that I am not using this tool for commercial purposes."*: `Yes`
+>    - *"Flexynesis utils"*: `Get Optimal Clusters`
```
Comment (pavanvidem) [inline topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md:193]: don't we have to run K-means clustering before this step? I am bit confused because there is also k-means clustering option in this tool.

File: topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md [hunk 9]
```diff
@@ -0,0 +1,303 @@
...
+> 1. {% tool [Flexynesis utils](toolshed.g2.bx.psu.edu/repos/bgruening/flexynesis_utils/flexynesis_utils/0.2.20+galaxy3) %} with the following parameters:
+>    - *"I certify that I am not using this tool for commercial purposes."*: `Yes`
+>    - *"Flexynesis utils"*: `Compute AMI and ARI`
+>        - {% icon param-file %} *"Predicted labels"*: `get_optimal_clusters` (output of **Flexynesis utils** {% icon tool %})
+>        - *"Column name in the labels file to use for the true labels"*: `c10` (celltype_l2)
+>        - *"Column name in the labels file to use for the predicted labels"*: `c13` (optimal_kmeans_cluster)
+{: .hands_on}
+> <question-title></question-title>
+>
+> 1. Which one of Louvain and k-means have better concordance?
```
Comment (pavanvidem) [inline topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md]:
```suggestion
> 1. Which of the clusterings has better concordance with the known cell type? Louvain and k-means?
```

File: topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md [hunk 10]
```diff
@@ -0,0 +1,303 @@
...
+>        - *"Minimum number of clusters to try"*: `10`
+>        - *"Maximum number of clusters to try"*: `20`
+>
+>    > <comment-title> Predicted labels </comment-title>
+>    >
+>    > Please make sure to use the output of Louvain clustering. We need those values in one table for next steps.
+>    {: .comment}
+>
+{: .hands_on}
+
```
Comment (pavanvidem) [inline topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md:206]: please rename the output (`get_optimal_clusters`) because it is used several times in the next steps. It will be easier to refer to.

File: topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md [hunk 11]
```diff
@@ -0,0 +1,303 @@
...
+## UMAP visualisation of true and Louvain lables
+
+> <hands-on-title> Dimension reduction plot </hands-on-title>
+>
+> 1. {% tool [Flexynesis plot](toolshed.g2.bx.psu.edu/repos/bgruening/flexynesis_plot/flexynesis_plot/0.2.20+galaxy3) %} with the following parameters:
+>    - *"I certify that I am not using this tool for commercial purposes."*: `Yes`
+>    - *"Flexynesis plot"*: `Dimensionality reduction`
+>        - {% icon param-file %} *"Predicted labels"*: `get_optimal_clusters` (Input dataset)
+>        - *"Column in the labels file to use for coloring the points in the plot"*: `c10` (celltype_l2)
```
Comment (pavanvidem) [inline topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md:269]: what about "Embeddings" param?

File: topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md [hunk 12]
```diff
@@ -0,0 +1,303 @@
...
+>        - *"Column in the labels file to use for coloring the points in the plot"*: `c12` (louvain_cluster)
+>        - *"Transformation method"*: `UMAP`
+{: .hands_on}
+
+> <question-title></question-title>
+>
+> 1. Compare these two UMAP plots, Is the unsupervised clustering close to the ground truth labels?
+>
+> > <solution-title></solution-title>
+> >
```
Comment (pavanvidem) [inline topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md:288]: can you please paste the plots here to support the solution?

File: topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md [hunk 13]
```diff
@@ -0,0 +1,303 @@
...
+>
+> 1. {% tool [Flexynesis plot](toolshed.g2.bx.psu.edu/repos/bgruening/flexynesis_plot/flexynesis_plot/0.2.20+galaxy3) %} with the following parameters:
+>    - *"I certify that I am not using this tool for commercial purposes."*: `Yes`
+>    - *"Flexynesis plot"*: `Label concordance heatmap`
+>        - {% icon param-file %} *"Predicted labels"*: `get_optimal_clusters` (output of **Flexynesis utils** {% icon tool %})
+>        - *"Column in the labels file to use for true labels"*: `c10` (celltype_l2)
+>        - *"Column in the labels file to use for predicted labels"*: `c12` (louvain_cluster)
+{: .hands_on}
+
+Now it is easier to see which Lovain cluster corresponds to which true value.
```
Comment (pavanvidem) [inline topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md:310]: Please paste the plot here and also explain a bit.

**Review by pavanvidem — COMMENTED** [review]
Thanks @nilchia
I also tested it and works well.

File: topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md (hunk 5 above)
Comment (nilchia) [reply topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md:26]: Done! thanks!

File: topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md (hunk 7 above)
Comment (nilchia) [reply topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md]: Yes, it is better to set to 5 to 15. I'll update that

File: topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md (hunk 8 above)
Comment (nilchia) [reply topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md:193]: The k-means clustering will only accept one k.
The optimal clustering chooses the best clustering from a range of K based on silhouette scores.

File: topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md (hunk 1 above)
Comment (nilchia) [reply topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md:60]: I tried to add it here [8917589](https://github.com/galaxyproject/training-material/pull/6270/commits/891758902492f2a1caddd98f59769e1818771245)

File: topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md (hunk 2 above)
Comment (nilchia) [reply topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md]: I tried to address it here[8f8e519](https://github.com/galaxyproject/training-material/pull/6270/commits/8f8e519f726dd9b01326b076bd98381871ecdc28)

File: topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md (hunk 6 above)
Comment (nilchia) [reply topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md:195]: Thanks, fixed here [5545a7a](https://github.com/galaxyproject/training-material/pull/6270/commits/5545a7a36e9101746cd1ee6ff37606b236ef0ab5)

File: topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md (hunk 10 above)
Comment (nilchia) [reply topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md:206]: Thanks, fixed here [5545a7a](https://github.com/galaxyproject/training-material/pull/6270/commits/5545a7a36e9101746cd1ee6ff37606b236ef0ab5)

File: topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md (hunk 11 above)
Comment (nilchia) [reply topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md:269]: Thanks, fixed here [5545a7a](https://github.com/galaxyproject/training-material/pull/6270/commits/5545a7a36e9101746cd1ee6ff37606b236ef0ab5)

File: topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md (hunk 12 above)
Comment (nilchia) [reply topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md:288]: Added here [c43e123](https://github.com/galaxyproject/training-material/pull/6270/commits/c43e123279569947598468607d01b4e7a806598b)

File: topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md (hunk 13 above)
Comment (nilchia) [reply topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md:310]: Added here [c43e123](https://github.com/galaxyproject/training-material/pull/6270/commits/c43e123279569947598468607d01b4e7a806598b)

File: topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md [hunk 14]
```diff
@@ -0,0 +1,317 @@
...
+> > 1. We can see that like true labels, each UMAP clusters have unique Louvain clusters assigned. This shows that this clustering based on the latent space is close to the ground truth. However, we still don't know which Louvain cluster, corresponds to which true label.
+> >
+> >![UMAP plot of test Embeddings colored by true labels](../../images/flexynesis_unsupervised/true_label_umap.jpeg "UMAP plot of test Embeddings colored by true labels")
+> >![UMAP plot of test Embeddings colored by predicted labels](../../images/flexynesis_unsupervised/predicted_label_umap.jpeg "UMAP plot of test Embeddings colored by predicted labels")
+> >
+> {: .solution}
+>
+{: .question}
+
+To see the real concordance between Louvain clusters and true values, we can bserve a tabulation of the concordance between them. (Each row sums up to 1).
```
Comment (bgruening) [inline topics/statistics/tutorials/flexynesis_unsupervised/tutorial.md]:
```suggestion
To see the real concordance between Louvain clusters and true values, we can observe a tabulation of the concordance between them. (Each row sums up to 1).
```

**Review by bgruening — APPROVED** [review]
